### PR TITLE
chore(pagination): move pagination logic to apollo-utils

### PR DESCRIPTION
**Pull request by sri-kirsh**
## Goal
We added the validatePagination logic to apollo-utils so it can be shared with repos like user-list-search.
In this PR, we are removing the `validatePagination` logic in list-api and using it from apollo-utils 

Ticket: 
[INFRA-244](https://getpocket.atlassian.net/browse/INFRA-244)

### CODE REVIEW

File: src/config/index.ts [hunk 1]
```diff
@@ -81,6 +81,6 @@ export default {
   parserVersion: process.env.PARSER_VERSION || 'v3beta',
   pagination: {
     defaultPageSize: 30,
-    maxPageSize: 100
-  }
+    maxPageSize: 100,
+  },
```
Comment (jpetto) [inline src/config/index.ts:85]: swooping in to note that updating the linter to our [latest standards](https://github.com/Pocket/collection-api/pull/618) may solve this.

**Review by jpetto — DISMISSED** [review]
this looks good to me! yay for abstracting logic.

File: src/resolvers/user.ts [hunk 2]
```diff
@@ -60,11 +63,13 @@ export function savedItems(
 export async function tags(
   parent: User,
   args: {
-    pagination: PaginationInput;
+    pagination: Pagination;
   },
   context: IContext
 ): Promise<TagConnection> {
-  validatePagination(args.pagination);
+  validatePagination(args.pagination,
```
Comment (sri-kirsh) [inline src/resolvers/user.ts:70]: todo: should assign it back to args.pagination (or) `assign pagination = {first:default} when null` inside the [pagination module ](https://github.com/Pocket/apollo-utils/blob/main/src/pagination/validatePagination.ts#L10)

File: src/resolvers/user.ts [hunk 3]
```diff
@@ -39,11 +40,13 @@ export function savedItems(
   args: {
     filter: SavedItemsFilter;
     sort: SavedItemsSort;
-    pagination: PaginationInput;
+    pagination: Pagination;
   },
   context: IContext
 ): Promise<SavedItemConnection> {
-  validatePagination(args.pagination);
+  validatePagination(args.pagination,
```
Comment (sri-kirsh) [inline src/resolvers/user.ts:47]: todo: should assign it back to args.pagination.

Comment (kkyeboah) [reply src/resolvers/user.ts:47]: objects are passed by reference in javascript, this might be ok/

Comment (sri-kirsh) [reply src/resolvers/user.ts:47]: ya thats what i thought too, but realized we have a logic in apollo-utils `when pagination== null, return {first:defaultSize}` and this won't get assigned back to our `args.pagination` 

https://github.com/Pocket/apollo-utils/blob/main/src/pagination/validatePagination.ts#L10

i can either do this or assign `pagination = {first: defaultSize`} in apollo-utils

**Review by jpetto — APPROVED** [review]
:shipit:

**Pull request merged**